### PR TITLE
Modify view underlying project listing page to address signal duplication

### PR DESCRIPTION
**Pull request by frankhereford**
## Associated issues

This PR intends to close [#9167](https://github.com/cityofaustin/atd-data-tech/issues/9167).

## Testing
**URL to test:** 

Testing requires a migration, so this needs to be on [_test_](https://moped-test.austinmobility.io/moped) or in your local development environment.

**Steps to test:**

* Create a new project using a signal asset so you start out with a component and feature
* Add at least one more signal component, selecting another signal. 
* [ ] 2+ features shown on map?
* Navigate to the summary page and add 2+ project partners
* Navigate to the funding tab and add 2+ task orders
* Navigate to the project listing page
* [ ] 2+ partners shown on the listing table, but only as many as you added?
* [ ] 2+ task orders shown on the listing table,(show the column first), but only as many as you added?
* [ ] 2+ signals shown on listing, but only the exact ones you added an no dups?

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-database/migrations/1652877867946_project-list-view-dedup-signal-assets/up.sql [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    mp.contractor,
+    mp.purchase_order_number,
+    COALESCE( -- coalese because this subquery can come back 'null' if there are no component assets
+     (SELECT JSON_AGG(features.feature) -- this query finds any components and those component's features and rolls them up in a JSON blob
+      FROM moped_proj_components components   
+     LEFT JOIN moped_proj_features features 
+       ON (features.project_component_id = components.project_component_id)
+     WHERE TRUE
+       and components.status_id = 1
+       and components.project_id = mp.project_id), '{}'::json) as project_feature, -- close out that coalese; if null, give us a empty json object
```
Comment (frankhereford) [inline moped-database/migrations/1652877867946_project-list-view-dedup-signal-assets/up.sql:66]: 🆕 Here's the bit that collects the component & feature JSON blob based on the project id of the row.

File: moped-database/migrations/1652877867946_project-list-view-dedup-signal-assets/down.sql [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+   FROM public.moped_project mp
+     LEFT JOIN project_person_list_lookup ppll ON mp.project_id = ppll.project_id
+     LEFT JOIN funding_sources_lookup fsl ON fsl.project_id = mp.project_id
+     LEFT JOIN project_type_lookup ptl ON ptl.project_id = mp.project_id
+     LEFT JOIN public.moped_entity me ON me.entity_id = mp.project_sponsor
+     LEFT JOIN public.moped_proj_partners mpp2 ON (mp.project_id = mpp2.project_id AND mpp2.status_id = 1)
+     LEFT JOIN public.moped_entity me2 ON mpp2.entity_id = me2.entity_id
+     LEFT JOIN jsonb_array_elements(mp.task_order) as task_order_filter ON true
+     LEFT JOIN moped_proj_components mpc ON mpc.project_id = mp.project_id and mpc.status_id = 1
+     LEFT JOIN moped_proj_features mpf ON mpc.project_component_id = mpf.project_component_id
```
Comment (frankhereford) [inline moped-database/migrations/1652877867946_project-list-view-dedup-signal-assets/down.sql:122]: 🔴 These two joins are removed

**Review by chiaberry — APPROVED** [review]
tested on moped-test and reviewed code. Thanks for a speedy fix!

**Review by patrickm02L — APPROVED** [review]
No more duplication of ID numbers. Good to go!

Screen Shot 2022-05-18 at 10 52 44 AM

**Pull request merged**